pkg/cluster/controllers/harbor/harbor.go

```diff
@@ -57,9 +57,11 @@ func (harbor *Controller) Apply(ctx context.Context, harborcluster *goharborv1.H
 		return harborUnknownStatus(GetHarborCRError, err.Error()), err
 	}
 
+	dep := checksum.New(harbor.Scheme)
+	dep.Add(ctx, harborcluster, true)
+
 	// Found the existing one and check whether it needs to be updated
-	same := equality.Semantic.DeepDerivative(desiredCR.Spec, harborCR.Spec)
-	if !same {
+	if !dep.ChangedFor(ctx, harborCR) {
```

Comment:
I suggest to write an helper function to make the 3 lines. It seems to be used many times.